spanner/api/Spanner/Program.cs

```diff
@@ -1888,6 +1943,267 @@ await cmd.ExecuteReaderAsync().ConfigureAwait(false))
+                { "LastName", SpannerDbType.String, "Campbell" },
+            };
+            // [END spanner_create_struct_with_data]
+
+            // [START spanner_query_data_with_struct]
+            string connectionString =
+            $"Data Source=projects/{projectId}/instances/"
+            + $"{instanceId}/databases/{databaseId}";
+            using (var connection = new SpannerConnection(connectionString))
+            {
+                using (var cmd = connection.CreateSelectCommand(
+                    "SELECT SingerId FROM Singers "
+                    + "WHERE STRUCT<FirstName STRING, LastName STRING>"
+                    + "(FirstName, LastName) = @name"))
+                {
+                    cmd.Parameters.Add("name", nameStruct.GetSpannerDbType(), nameStruct);
+                    using (var reader = await cmd.ExecuteReaderAsync())
+                    {
+                        while (await reader.ReadAsync())
+                        {
+                            Console.WriteLine(
+                                reader.GetFieldValue<string>("SingerId"));
+                        }
+                    }
+                }
+            }
+            // [END spanner_query_data_with_struct]
+        }
+
+        public static object QueryDataWithArrayOfStruct(
+            string projectId, string instanceId, string databaseId)
+        {
+            var response = QueryDataWithArrayOfStructAsync(
+                projectId, instanceId, databaseId);
+            s_logger.Info("Waiting for operation to complete...");
+            response.Wait();
+            s_logger.Info($"Response status: {response.Status}");
+            return ExitCode.Success;
+        }
+
+        public static async Task QueryDataWithArrayOfStructAsync(
+            string projectId, string instanceId, string databaseId)
+        {
+            // [START spanner_create_user_defined_struct]
+            var nameType = new SpannerStruct {
+                { "FirstName", SpannerDbType.String, null},
+                { "LastName", SpannerDbType.String, null}
+            };
+            // [END spanner_create_user_defined_struct]
+
+            // [START spanner_create_array_of_struct_with_data]
+            var bandMembers = new List<SpannerStruct>
+            {
+                new SpannerStruct { { "FirstName", SpannerDbType.String, "Elena" },
+                    { "LastName", SpannerDbType.String, "Campbell" } },
+                new SpannerStruct { { "FirstName", SpannerDbType.String, "Gabriel" },
+                    { "LastName", SpannerDbType.String, "Wright" } },
+                new SpannerStruct { { "FirstName", SpannerDbType.String, "Benjamin" },
+                    { "LastName", SpannerDbType.String, "Martinez" } },
+            };
+            // [END spanner_create_array_of_struct_with_data]
+
+
+            // [START spanner_query_data_with_array_of_struct]
+            string connectionString =
```

Comment:
I'd prefer it to be as you suggest too. Since this is code to be included in documentation (which has width limitations) and it mirrors the existing approach already in the sample. I'm going to opt to leave as it for this PR.